relay-server/src/actors/store.rs

```diff
@@ -408,6 +422,54 @@ impl StoreForwarder {
         );
         Ok(())
     }
+
+    fn produce_profiling_session_message(&self, item: &Item) -> Result<(), StoreError> {
+        let message = ProfilingKafkaMessage {
+            organization_id: item
+                .get_header("organization_id")
```

Comment:
This will panic if the header is missing

Reply:
Generally I believe the SDK should not send any organization ID at all, and it should not be in the item header. It should be inferred from the actual DSN that the SDK uses to authenticate with, due to:

1. security reasons, SDKs should not determine org ID freely after authenticating with any DSN
2. ergonomics, SDK should not have to be configured with more than DSN to start sending data successfully

Check how other methods do it, they access envelope scoping to get information about org/project.

Reply:
I agree with this. We should fill in the org ID in relay. I also wonder if we need this installation_id, is that not the project_id we get from the public key anyways?

Reply:
if installation id is also a customer identifier it needs to go away as well imo

Reply:
`installation_id` was just a way for us to count the number of devices sending traces. It has no use besides that, we can remove it.

`organization_id` was meant as a temporary measure so we can match a Sentry ID with a Specto ID (this organization ID in the headers would represent the Specto ID, not Sentry), but since we're not going to release our SDK to existing Specto users, we can remove that and just maintain a mapping of IDs on our side.

No ID, no problem.

Reply:
How does specto service know either ID then? Is it in the payload? I don't see it in the Kafka message then.

Reply:
@phacops and I had a chat on that, and it depends on how much orgs and projects are integrated during Phase 1. 
- If Specto requires their proprietary IDs, then they can provide them through the header for now, and we'll back it out during Phase 2. We can consider all of this API experimental. 
- If Specto can deal can map our IDs on their end, we would add this directly to the Kafka message in Relay, just like we do for Metrics or Sessions. @phacops if you trace the callers of [this](https://github.com/getsentry/relay/blob/d148b24f8e9d64e4e705be6157843dc41a64fa60/relay-server/src/actors/store.rs#L278-L279), you'll see how to pull that off.

> I don't see it in the Kafka message then.

+1. Do you intend to add this as headers to the Kafka message, or include it in its payload?

Reply:
For the `organization_id`, I added Sentry's ID to the Kafka message (https://github.com/getsentry/relay/pull/1170/files#diff-b043940a406beb64d66fa208a834b04f23942ae398a6cbe8633514e9ddff0eabR432) and we'll map it on our end. In phase 1, only the Specto organization will send traces, so that's easy and starting phase 2, we'll only use Sentry's IDs.

We have an `app_id` inside the `payload`, which I think would app with your `project_id`. I don't know if changing this right now is a good idea, phase 1 is really about showing something in the Sentry's UI and it's simpler to not change it right now to get to that goal. I can add the `project_id` to the message but likely won't use it for now.

Reply:
It turns out, we can also forget our `app_id` for now and just use the `project_id`, even for phase 1. So, I think we're all good now since we don't have custom/private IDs anymore.

Reply:
>We have an app_id inside the payload, which I think would app with your project_id. I don't know if changing this right now is a good idea, phase 1 is really about showing something in the Sentry's UI and it's simpler to not change it right now to get to that goal.

If I understand you correctly you moved on from this approach, but when this is deployed we have potentially this problem:

1. sdk authenticates as org X via DSN, and sends data under app ID belonging to org Y
2. specto stores data under org Y even though user only had credentials to X

if you think of `installation_id` as a credential then everything is fine again, but I would argue that needing two credentials is not ideal and the added complexity adds risk. I would hope that Relay can correctly authenticate all Specto traffic before we deploy anything, and there's no unauthenticated data on the kafka topic.

I'm glad you found a way to get rid of the specto IDs!

Reply:
The `installation_id` is not some sort of credential, it was purely to distinguish between 2 installations of Specto. That's why we can remove it now. The `app_id` inside the payload won't be used anymore, we'll use the `project_id` instead. I think this solves the potential problem you're talking about.

Also, keep in mind this is for an internal release and the Specto SDK will be scraped during phase 2 as we transition to the Sentry SDK.